File: go/difference-of-squares/difference_of_squares.go
package diffsquares

func SquareOfSums(n int) int {
	sum := (n * (n + 1)) / 2
	return sum * sum
}

func SumOfSquares(n int) (acc int) {
	for i := 1; i <= n; i++ {
		acc += i * i
	}
    return
}

func Difference(n int) int {
	return SquareOfSums(n) - SumOfSquares(n)
}

File: erlang/difference-of-squares/difference_of_squares.erl
-module(difference_of_squares).
-export([sum_of_squares/1, square_of_sums/1, difference_of_squares/1]).

square(N) -> trunc(N * N).

sum_of_squares(N) ->
    Squares = lists:map(fun square/1, lists:seq(1, N)),
    lists:foldl(fun (S, Sum) -> S + Sum end, 0, Squares).

square_of_sums(N) ->
    Sum = (N * (N + 1)) / 2,
	square(Sum).

difference_of_squares(N) ->
    square_of_sums(N) - sum_of_squares(N).
